commit c5b8e20d0ec380c2f2222fa266261d22dc36f926
Author: Mzhelskiy Maxim <rus.engine@gmail.com>
Date:   Sun Jan 8 06:34:45 2012 +0400

    fix XSS bug

diff --git a/classes/actions/ActionPhotoset.class.php b/classes/actions/ActionPhotoset.class.php
index 0d4732d..a498f48 100644
--- a/classes/actions/ActionPhotoset.class.php
+++ b/classes/actions/ActionPhotoset.class.php
@@ -337,6 +337,10 @@ class ActionPhotoset extends Action {
 		 * Устанавливаем шаблон вывода
 		 */
 		$this->SetTemplateAction('add');
+
+		if (!is_numeric(getRequest('topic_id'))) {
+			$_REQUEST['topic_id']='';
+		}
 		/**
 		 * Проверяем отправлена ли форма с данными(хотяб одна кнопка)
 		 */		
@@ -384,6 +388,10 @@ class ActionPhotoset extends Action {
 		$this->Viewer_Assign('aBlogsAllow',$this->Blog_GetBlogsAllowByUser($this->oUserCurrent));
 		$this->Viewer_AddHtmlTitle($this->Lang_Get('topic_photoset_title_create'));
 
+		if (!is_numeric(getRequest('topic_id'))) {
+			$_REQUEST['topic_id']='';
+		}
+
 		// Если нет временного ключа для нового топика, то генерируеи. если есть, то загружаем фото по этому ключу
 		if (empty($_COOKIE['ls_photoset_target_tmp'])) {
 			setcookie('ls_photoset_target_tmp',  func_generator(), time()+24*3600,Config::Get('sys.cookie.path'),Config::Get('sys.cookie.host'));
diff --git a/templates/skin/developer-jquery/actions/ActionPhotoset/add.tpl b/templates/skin/developer-jquery/actions/ActionPhotoset/add.tpl
index 5de1bba..70d524c 100644
--- a/templates/skin/developer-jquery/actions/ActionPhotoset/add.tpl
+++ b/templates/skin/developer-jquery/actions/ActionPhotoset/add.tpl
@@ -53,7 +53,7 @@ tinyMCE.init({
 <script type="text/javascript">
 	if (jQuery.browser.flash) {
 		ls.photoset.initSwfUpload({
-			post_params: { 'topic_id':'{$_aRequest.topic_id}' }
+			post_params: { 'topic_id': {json var=$_aRequest.topic_id} }
 		});
 	}
 </script>
diff --git a/templates/skin/developer/actions/ActionPhotoset/add.tpl b/templates/skin/developer/actions/ActionPhotoset/add.tpl
index 70d9d03..014d7d2 100644
--- a/templates/skin/developer/actions/ActionPhotoset/add.tpl
+++ b/templates/skin/developer/actions/ActionPhotoset/add.tpl
@@ -60,7 +60,7 @@ tinyMCE.init({
 <script type="text/javascript">
 	if (Browser.Plugins.Flash.version) {
 		initSwfUpload({
-			post_params: { 'topic_id':'{$_aRequest.topic_id}' },
+			post_params: { 'topic_id': {json var=$_aRequest.topic_id} },
 			events: {
 				UploadProgress: swfHandlerUploadProgress,
 				FileDialogComplete: swfHandlerFileDialogComplete,
diff --git a/templates/skin/new-jquery/actions/ActionPhotoset/add.tpl b/templates/skin/new-jquery/actions/ActionPhotoset/add.tpl
index 813822b..f269c1f 100644
--- a/templates/skin/new-jquery/actions/ActionPhotoset/add.tpl
+++ b/templates/skin/new-jquery/actions/ActionPhotoset/add.tpl
@@ -52,7 +52,7 @@ tinyMCE.init({
 <script type="text/javascript">
 if (jQuery.browser.flash) {
 	ls.photoset.initSwfUpload({
-		post_params: { 'topic_id':'{$_aRequest.topic_id}' }
+		post_params: { 'topic_id': {json var=$_aRequest.topic_id} }
 	});
 }
 </script>
diff --git a/templates/skin/new/actions/ActionPhotoset/add.tpl b/templates/skin/new/actions/ActionPhotoset/add.tpl
index 1c4c26a..3fdc363 100644
--- a/templates/skin/new/actions/ActionPhotoset/add.tpl
+++ b/templates/skin/new/actions/ActionPhotoset/add.tpl
@@ -60,7 +60,7 @@ tinyMCE.init({
 <script type="text/javascript">
 	if (Browser.Plugins.Flash.version) {
 		initSwfUpload({
-			post_params: { 'topic_id':'{$_aRequest.topic_id}' },
+			post_params: { 'topic_id': {json var=$_aRequest.topic_id} },
 			events: {
 				UploadProgress: swfHandlerUploadProgress,
 				FileDialogComplete: swfHandlerFileDialogComplete,
